### PR TITLE
fix: Parse unidentified shiny items correctly [skip ci]

### LINKED ISSUE #2488: Unidentified Shiny Items are not annotated correctly
GearItem currently expects the shiny symbol to be before the "Unidentified".

image

### PR DESCRIPTION
**Pull request by kristofbolyai**
I think this is all we need to fix https://github.com/Wynntils/Artemis/issues/2488.

image

### CODE REVIEW

**Comment by kristofbolyai** [comment]
Actually, it's not how it works..

File: common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
 import net.minecraft.world.item.ItemStack;
 
 public final class GearAnnotator implements GameItemAnnotator {
-    private static final Pattern GEAR_PATTERN =
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java:19]: Maybe add a regexp test?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java:19]: My gradle setup is slightly broken and I can't run tests, and I have yet to fix that...

**Comment by ShadowCat117** [comment]
Doesn't work for id'd shinies and marks them as common. I think that's the solution I tried and ended up with this result
image

**Comment by kristofbolyai** [comment]
I'll just separate the unid and not unid regexes then.

**Comment by ShadowCat117** [comment]
Unfortunately I think that's the only solution :/

**Comment by kristofbolyai** [comment]
Actually, I think I can do 1 pattern.

**Comment by kristofbolyai** [comment]
@JamieCallan117 Should work now.

File: common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java:19]: Oh, that's really bad. :( I think this is a case where a test really would be beneficial, especially seeing as how this has been going back and forth with different strings that failed in one way or another.

I had a similar problem with a dev environment that I had customized yet were broken in some way. What I did was just clone the repo to another dir and open as a separate project in IntelliJ. I recommend you do that, and just run the test.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java:19]: I just pushed the tests, and if they fail, we will see it on the github workflow.

**Review by ShadowCat117 — APPROVED** [review]
Looks good, just add the `// Test in GearAnnotator_GEAR_PATTERN` and it's good to go

**Comment by kristofbolyai** [comment]
> Just add the `// Test in GearAnnotator_GEAR_PATTERN` and it's good to go

We still do that?

**Comment by ShadowCat117** [comment]
I think all of the other test cases have it

**Comment by magicus** [comment]
I'm not quite sure why we still do that, but I think we argued that it helps folks remember to update tests, since we are not really into the habit of always running tests, or that all regexes has tests to it is second nature to always update them.

**Review by magicus — APPROVED** [review]
And thanks for adding the test!

**Pull request merged**